sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreCoordinator.scala

```diff
@@ -55,6 +56,25 @@ private case class GetLocation(storeId: StateStoreProviderId)
 private case class DeactivateInstances(runId: UUID)
   extends StateStoreCoordinatorMessage
 
+private case class SnapshotUploaded(storeId: StateStoreProviderId, version: Long, timestamp: Long)
+  extends StateStoreCoordinatorMessage
+
+/**
+ * Message used for testing.
+ * This message is used to retrieve the latest snapshot version reported for upload from a
+ * specific state store instance.
+ */
+private case class GetLatestSnapshotVersion(storeId: StateStoreProviderId)
+  extends StateStoreCoordinatorMessage
+
+/**
+ * Message used for testing.
+ * This message is used to retrieve the all active state store instance falling behind in
+ * snapshot uploads, whether it is through version or time criteria.
+ */
+private case class GetLaggingStores()
```

Comment:
nit: add `ForTesting` suffix to method name, so it is super obvious it is only for testing